### PR TITLE
feat: improved release of database resources.

### LINKED ISSUE #77: Improve resource management for block repository backends
It may be needed to allow for callers to manage resources when using the map data from the block repository. I see two opportunities for improvement/fixes:

1/ Implement defer Rows.Close() and DB.Close() on all backends: this may solve memory usage issues as reported here https://github.com/minetest-go/mapcleaner/issues/179 as well as resolve final transactions/connections not being properly released (potentially what is happening here: https://github.com/minetest-go/mapcleaner/issues/180.

2/ Implement a new version of Iterator(), IteratorTo() which receives a block bounding box. It could make concurrency easier to implement like for exporting all protected areas in separate go-routines. Also, perhaps a single iterator that keeps sql.Rows always open and only closes at the end could be a bad idea. If there are just too many blocks in the map (the main use case for cleaning the map) it can use just too much memory.

Open for feedback/thoughts on this!

----

Checklist:

- [x] Implement BlockRepository.Close().
- [x] Ensure sql.Rows.Close() is called.
- [ ] Batch Iterator() to avoid single sql.Rows open for too much time.
- [ ] #79

### PR DESCRIPTION
**Pull request by ronoaldo**
- Added sql.Rows.Close() calls when missing
- Allows the BlockRepository to be closed, and thus closing
  the underlying sql.DB connections.

Fixes #77

### CODE REVIEW

**Comment by ronoaldo** [comment]
This indeed seems to perhaps fix the issues! I was trying to reproduce https://github.com/minetest-go/mapcleaner/issues/179 but when testing with PostgreSQL backend it was not happening. It seems to be caused by the missing Rows.Close() holding memory.

Also, after implementing defer to Close() the BlockRepository, I was able to test and mapcleanner no longer leaves dangling WAL related files for sqlite.

Worth mergning and releasing to unblock https://github.com/minetest-go/mapcleaner/pull/183

**Review by BuckarooBanzay — APPROVED** [review]
lgtm (i was _pretty_ sure i called the `Close()` func everywhere but :shrug:)

**Comment by ronoaldo** [comment]
> lgtm (i was _pretty_ sure i called the `Close()` func everywhere but 🤷)

I managed to review/fix this only for one backend too 🙃.

**Pull request merged**
